src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -1103,9 +1103,8 @@ private int GetDepthInMainChain(UtxoStakeDescription utxoStakeDescription)
             ChainedHeader chainedBlock = this.chain.GetBlock(utxoStakeDescription.HashBlock);
 
             if (chainedBlock == null)
-                return -1;
+                return this.mempool.Exists(utxoStakeDescription.UtxoSet.TransactionId) ? 0 : -1;
```

Comment:
Ah good spot, currently access to mempool has to be protected with the mempool lock.
```
 await this.mempoolLock.ReadAsync(() => this.mempool.Exists(utxoStakeDescription.UtxoSet.TransactionId) ? 0 : -1).ConfigureAwait(false);
```

We should probably refactor the mempool lock in any case.